cloudinit/util.py

```diff
@@ -1255,19 +1255,44 @@ def close_stdin():
         os.dup2(fp.fileno(), sys.stdin.fileno())
 
 
+def find_devs_with_freebsd(criteria=None, oformat='device',
```

Comment:
why are we making the `criteria` parameter into an ad-hoc parser, rather than having `type` and `label` parameters (defaulting to `None`).

Reply:
oh… because it's an existing function we're extending and we're just making it compatible…

Reply:
Yes, absolutely. This being said, since we call it ourselves, we can probably also change the functions API. I will take a look at that. Thank @igalic for your feedback.

Reply:
`criteria` is an opaque string in `find_devs_with` because it just gets passed to `blkid` verbatim on Linux: https://github.com/canonical/cloud-init/blob/9efa64f5a7142466a70df805787a4ab347fd9eba/cloudinit/util.py#L1312-L1319
(Not saying you shouldn't modify the signature of the non-Linux functions, just giving some background!)